core-web/libs/sdk/client/src/lib/editor/sdk-editor.ts

```diff
@@ -93,26 +97,44 @@ export function reorderMenu(config?: ReorderMenuConfig): void {
 }
 
 /**
- * Checks if the code is running inside an editor.
+ * Checks if the code is running inside the DotCMS editor.
+ *
+ * @param {InsideEditorOptions} [options] - Optional configuration options
+ * @param {boolean} [options.checkPreview=false] - If true, checks if running in preview mode and returns false if so
+ * @returns {boolean} Returns true if running inside the editor, false otherwise
+ *
+ * @description
+ * This function determines if the code is executing within the DotCMS editor by performing these checks:
+ * 1. Verifies if window exists (returns false if server-side rendering)
+ * 2. If checkPreview is true, checks if in preview mode and returns false if so
+ * 3. Checks if running inside an iframe by comparing window.parent to window
+ *
+ * The main use case is to conditionally execute code only when running inside the DotCMS editor.
+ * By default it will return true even in preview mode, but you can pass checkPreview: true to
+ * exclude preview mode.
  *
- * @returns {boolean} Returns true if the code is running inside an editor, otherwise false.
  * @example
  * ```ts
+ * // Basic check if running in editor (including preview mode)
  * if (isInsideEditor()) {
- *     console.log('Running inside the editor');
- * } else {
- *     console.log('Running outside the editor');
+ *     // Code will run in both edit and preview modes
+ * }
+ *
+ * // Check if in editor but NOT in preview mode
+ * if (isInsideEditor({ checkPreview: true })) {
+ *     // Code will only run in edit mode
  * }
  * ```
  */
-export function isInsideEditor(): boolean {
+export function isInsideEditor(
+    { checkPreview }: InsideEditorOptions = { checkPreview: false }
```

Comment:
why?

Reply:
This was to (in a nutshell) know if we want to render something while in the editor in Preview mode

Now this logic has been moved to a function called detectUVEContext, so as not to touch the current operation of isInsideEditor